pkg/hintrunner/hintparser.go

```diff
@@ -0,0 +1,297 @@
+package hintrunner
+
+import (
+	"fmt"
+)
+
+// Possible cases extracted from https://github.com/lambdaclass/cairo-vm_in_go/blob/main/pkg/hints/hint_utils/hint_reference.go#L41
+// Immediate: cast(number, type)
+// Reference no deref 1 offset: cast(reg + off, type)
+// Reference no deref 2 offsets: cast(reg + off1 + off2, type)
+// Reference with deref 1 offset: cast([reg + off1], type)
+// Reference with deref 2 offsets: cast([reg + off1] + off2, type)
+// Two references with deref: cast([reg + off1] + [reg + off2], type)
+// Reference off omitted: cast(reg, type)
+// Reference with deref off omitted: cast([reg], type)
+// Reference with deref 2 offsets off1 omitted: cast([reg] + off2, type)
+// 2 dereferences off1 omitted: cast([reg] + [reg + off2], type)
+// 2 dereferences off2 omitted: cast([reg + off1] + [reg], type)
+// 2 dereferences both offs omitted: cast([reg] + [reg], type)
+// Reference no dereference 2 offsets - + : cast(reg - off1 + off2, type)
+
+// Note: The same cases apply with an external dereference. Example: [cast(number, type)]
+
+type IdentifierExp struct {
+	DerefCastExp *DerefCastExp `@@ |`
+	CastExp      *CastExp      `@@`
+}
+
+type DerefCastExp struct {
+	CastExp *CastExp `"[" @@ "]"`
+}
+
+type CastExp struct {
+	ValueExpr *Expression `"cast" "(" @@ ","`
+	CastType  []string  `@Ident ("." @Ident)* ("*")? ("*")? ")"`
+}
+
+type Expression struct {
+	CellRefExp *CellRefExp `"(" @@ ")" | @@ |`
+	DerefExp   *DerefExp   `@@ |`
+	BinOpExp   *BinOpExp   `@@`
+}
+
+type CellRefExp struct {
+	RegisterOffset *RegisterOffset `@@ |`
+	Register       string          `@("ap" | "fp")`
+}
+
+type RegisterOffset struct {
+	Register string     `@("ap" | "fp")`
+	Operator string     `@("+" | "-")`
+	Offset   *OffsetExp `@@`
+}
+
+type DerefExp struct {
+	CellRefExp *CellRefExp `"[" @@ "]"`
+}
+
+type BinOpExp struct {
+	LeftExp  *LeftExp   `@@ "+"`
+	RightExp *RightExp  `@@`
+}
+
+type OffsetExp struct {
+	Number    *int `@Int |`
+	NegNumber *int `"(" "-" @Int ")"`
+}
+
+type LeftExp struct {
+	CellRefExp *RegisterOffset `@@ |`
+	DerefExp   *DerefExp   `@@`
+}
+
+type RightExp struct {
+	DerefExp *DerefExp  `@@ |`
+	Offset   *OffsetExp `@@`
+}
+
+type DerefOffset struct {
+	Deref  Deref
+	Offset *int
+}
+type DerefDeref struct {
+	LeftDeref  Deref
+	RightDeref Deref
+}
+
+
+// AST Functionality
+func (expression IdentifierExp) Evaluate() (interface{}, error) {
+	switch {
+	case expression.DerefCastExp != nil:
+		return expression.DerefCastExp.Evaluate()
+	case expression.CastExp != nil:
+		return expression.CastExp.Evaluate()
+	default:
+		return nil, fmt.Errorf("unexpected identifier value")
+	}
+}
+
+func (expression DerefCastExp) Evaluate() (interface{}, error) {
```

Comment:
I'm almost sure that we use both `any` and `interface{}` in our code base.
What do you think about sticking to `any`?
It's a minor thing, but `interface{}` looks less pretty even if it was the only option before `any` was introduced. :D

Reply:
We mostly use `interface{}` in the code base. But I agree with using `any` instead of `interface{}`. I've never liked `interface{}` in the first place. However, if we are sticking to `any` we should probably change it everywhere so as to keep the same format throughout the project.